Index: llvm-toolchain-3.9-3.9~svn275918/compiler-rt/test/lit.common.cfg
===================================================================
--- llvm-toolchain-3.9-3.9~svn275918.orig/compiler-rt/test/lit.common.cfg
+++ llvm-toolchain-3.9-3.9~svn275918/compiler-rt/test/lit.common.cfg
@@ -148,7 +148,7 @@ def is_linux_lto_supported():
   if not os.path.exists(os.path.join(config.llvm_shlib_dir, 'LLVMgold.so')):
     return False
 
-  ld_cmd = subprocess.Popen([config.gold_executable, '--help'], stdout = subprocess.PIPE)
+  ld_cmd = subprocess.Popen([config.gold_executable, '--help'], stdout = subprocess.PIPE, env={'LANG': 'C'})
   ld_out = ld_cmd.stdout.read().decode()
   ld_cmd.wait()
 
