Index: llvm-toolchain-snapshot_4.0~svn279916/polly/lib/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_4.0~svn279916.orig/polly/lib/CMakeLists.txt
+++ llvm-toolchain-snapshot_4.0~svn279916/polly/lib/CMakeLists.txt
@@ -1,10 +1,10 @@
 set(LLVM_NO_RTTI 1)
 
-set(POLLY_JSON_FILES
-    JSON/json_reader.cpp
-    JSON/json_value.cpp
-    JSON/json_writer.cpp
-)
+# set(POLLY_JSON_FILES
+#     JSON/json_reader.cpp
+#     JSON/json_value.cpp
+#     JSON/json_writer.cpp
+# )
 
 set(ISL_CODEGEN_FILES
     CodeGen/IslAst.cpp
@@ -50,7 +50,7 @@ add_polly_library(Polly
   Support/RegisterPasses.cpp
   Support/ScopHelper.cpp
   Support/ScopLocation.cpp
-  ${POLLY_JSON_FILES}
+#  ${POLLY_JSON_FILES}
   Transform/Canonicalization.cpp
   Transform/CodePreparation.cpp
   Transform/DeadCodeElimination.cpp
@@ -62,7 +62,7 @@ if (GPU_CODEGEN)
   target_link_libraries(Polly PollyPPCG)
 endif (GPU_CODEGEN)
 
-target_link_libraries(Polly PollyISL)
+target_link_libraries(Polly PollyISL jsoncpp)
 
 if (BUILD_SHARED_LIBS)
   target_link_libraries(Polly
Index: llvm-toolchain-snapshot_4.0~svn279916/polly/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_4.0~svn279916.orig/polly/CMakeLists.txt
+++ llvm-toolchain-snapshot_4.0~svn279916/polly/CMakeLists.txt
@@ -147,10 +147,13 @@ if (CUDALIB_FOUND)
   INCLUDE_DIRECTORIES( ${CUDALIB_INCLUDE_DIR} )
 endif(CUDALIB_FOUND)
 
+find_path(JSONCPP_INCLUDE_PATH json/json.h PATH_SUFFIXES jsoncpp)
+
 include_directories(
   BEFORE
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  ${CMAKE_CURRENT_SOURCE_DIR}/lib/JSON/include
+#  ${CMAKE_CURRENT_SOURCE_DIR}/lib/JSON/include
+  ${JSONCPP_INCLUDE_PATH}
   ${CMAKE_CURRENT_BINARY_DIR}/lib/External/isl/include
   ${CMAKE_CURRENT_SOURCE_DIR}/lib/External/isl/include
   ${CMAKE_CURRENT_SOURCE_DIR}/lib/External/pet/include
