Index: llvm-toolchain-snapshot_3.9~svn269409/clang/tools/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_3.9~svn269409.orig/clang/tools/CMakeLists.txt
+++ llvm-toolchain-snapshot_3.9~svn269409/clang/tools/CMakeLists.txt
@@ -16,6 +16,7 @@ endif()
 if(CLANG_ENABLE_STATIC_ANALYZER)
   add_clang_subdirectory(clang-check)
   add_clang_subdirectory(scan-build)
+  add_clang_subdirectory(scan-build-py)
   add_clang_subdirectory(scan-view)
 endif()
 
